gcloud/datastore/client.py

```diff
@@ -23,27 +23,32 @@
 from gcloud.datastore.key import Key
 from gcloud.datastore.query import Query
 from gcloud.datastore.transaction import Transaction
+from gcloud.datastore._implicit_environ import _determine_default_dataset_id
+from gcloud.datastore._implicit_environ import get_connection
 
 
 class Client(object):
     """Convenience wrapper for invoking APIs/factories w/ a dataset ID.
 
     :type dataset_id: string
-    :param dataset_id: (required) dataset ID to pass to proxied API methods.
-
-    :type namespace: string
-    :param namespace: (optional) namespace to pass to proxied API methods.
+    :param dataset_id: (optional) dataset ID to pass to proxied API methods.
 
     :type connection: :class:`gcloud.datastore.connection.Connection`, or None
     :param connection: (optional) connection to pass to proxied API methods
+
+    :type namespace: string
+    :param namespace: (optional) namespace to pass to proxied API methods.
     """
 
-    def __init__(self, dataset_id, namespace=None, connection=None):
+    def __init__(self, dataset_id=None, connection=None, namespace=None):
```

Comment:
`connection` is something a user should seldom / never pass, so we should keep it towards the end (I agree with @jgeewax on that).


I also agree that "examples just use keyword-style spellings, anyway"